enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,426 @@
+---
+title: dummy-gateway-interface-ovn
+authors:
+- "@pliurh"
+reviewers:
+- "@zshi-redhat, MicroShift"
+approvers:
+- "@dhellmann"
+api-approvers:
+- Node
+creation-date: 2023-04-26
+last-updated: 2023-04-26
+tracking-link:
+- https://issues.redhat.com/browse/NP-654
+see-also:
+- Node
+replaces:
+- Node
+superseded-by:
+- Node
+---
+
+# Single Node OVN-Kubernetes for Microshift
+
+## Summary
+
+This enhancement proposes a new external gateway configuration of OVN-Kubernetes
+for Microshift. With the new configuration, a dummy interface will be created
+and the `br-ex` bridge will use it as the gateway interface.
+
+## Motivation
+
+Micoshift is designed to run on IoT and edge devices, it can be deployed on
+various kinds of devices with different network environment. 
+
+Today configure-ovs.sh (microshift-ovs-init.service) sets up OVN-Kubernetes
+gateway bridge `br-ex` for OVN-Kubernetes.  There has been several types of
+issues associated with it mainly due to:
+
+1. transient network interruption when it moves the node IP from uplink to
+   gateway bridge
+
+2. certain type of network devices are not supported (e.g. wifi, tap/tun)
+
+3. it relies on (default) route info
+
+### User Stories
+
+* As an Microshift cluster administrator, I want the installation experience of
+Microshift is similar as a normal application. I don't want to have any
+networking interrupt during the installation.
+
+* As an Microshift cluster administrator, I want to use a specific network
+  devices (such as wifi, tap/tun) as the node default interface.
+
+### Goals
+
+* Decouple the br-ex bridge from the host physical interfaces.
+
+### Non-Goals
+
+* Creating a special gateway mode in OVN-Kubernetes for single node cluster.
+
+## Proposal
+
+Instead of creating the br-ex bridge with the node's default interface, we
+create a dummy interface and bind use it as the uplink of the br-ex. Since there
+is no real next-hop available via the dummy0 interface, we need to use the
```

Comment:
Bikeshed: Do we want to use the actual name "dummy0"? Is that name common to OVN-K, or networking generally? Would it make sense to use a name that indicates what it is for (maybe "microshift-internal", although that is long), or is "dummy" a convention?

Reply:
We can use whatever name we want. I use 'dummy0' to indicate this is a dummy interface.